### PR TITLE
Deprecation warning for deprecated types

### PR DESCRIPTION
**Pull request by esnible**
Resolves https://github.com/istio/istio/issues/5668

### CODE REVIEW

File: istioctl/cmd/istioctl/main.go [hunk 1]
```diff
@@ -97,6 +97,13 @@ var (
 		model.EgressRule.Type:        20,
 	}
 
+	// deprecatedTypes tracks if a deprecation warning is needed
+	deprecatedTypes = map[string]bool{
+		model.RouteRule.Type:         true,
+		model.IngressRule.Type:       true,
+		model.DestinationPolicy.Type: true,
```
Comment (ijsnellf) [inline istioctl/cmd/istioctl/main.go]: do we want EgressRule here as well? cc @vadimeisenbergibm

File: istioctl/cmd/istioctl/main.go [hunk 2]
```diff
@@ -171,6 +178,9 @@ See https://istio.io/docs/reference/ for an overview of Istio routing.
 					return err
 				}
 				var rev string
+				if deprecated, _ := deprecatedTypes[config.Type]; deprecated {
+					c.Printf("Warning: %s is deprecated and may not be supported in future Istio versions (%s).\n", config.Type, config.Name)
```
Comment (ayj) [inline istioctl/cmd/istioctl/main.go]: s/may/will 

Also, any chance we cab give a hard deprecation date/version here? Is 1.0 too soon?

File: istioctl/cmd/istioctl/main.go (hunk 1 above)
Comment (esnible) [reply istioctl/cmd/istioctl/main.go]: Good idea; I added it.

**Pull request merged**